azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/service/impl/AppServiceUtils.java

```diff
@@ -173,4 +179,31 @@ public static AppServicePlanEntity getAppServicePlanEntity(com.azure.resourceman
                 .operatingSystem(getOperatingSystem(appServicePlan.operatingSystem()))
                 .build();
     }
+
+    static AppServicePlan getAppServicePlan(AppServicePlanEntity entity, AzureResourceManager azureClient) {
+        try {
+            return StringUtils.isNotEmpty(entity.getId()) ?
+                    azureClient.appServicePlans().getById(entity.getId()) :
+                    azureClient.appServicePlans().getByResourceGroup(entity.getResourceGroup(), entity.getName());
+        } catch (Exception e) {
```

Comment:
change to an azure exception

Reply:
for example: when the network is down, the exception will be thrown, then do you think this plan doesn't exist?